test/integration/util.go

```diff
@@ -26,18 +26,21 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
-func waitFor(ctx context.Context, interval time.Duration, check func(chan error)) error {
-	ch := make(chan error, 1)
-	go func() {
-		for {
-			check(ch)
+func waitFor(ctx context.Context, interval time.Duration, check func() (bool, error)) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			// check functions return both a boolean and error to demonstrate
+			// whether an error has occurred and whether the function should
+			// continue to be called. waitFor will only exit when the check
+			// function returns true.
+			if done, err := check(); done {
+				return err
+			}
+			time.Sleep(interval)
```

Comment:
https://golang.org/pkg/time/#Ticker

Could you use a ticker instead? I think it's functionally equivalent but perhaps slightly more idiomatic.

Reply:
Yep, will update this :)